_api-reference/ingest-apis/ip2geo.md

```diff
@@ -0,0 +1,245 @@
+
+```json
+{
+  "datasources": [
+    {
+      "name": "my-datasource",
+      "state": "AVAILABLE",
+      "endpoint": "https://geoip.maps.opensearch.org/v1/geolite2-city/manifest.json",
+      "update_interval_in_days": 3,
+      "next_update_at_in_epoch_millis": 1685125612373,
+      "database": {
+        "provider": "maxmind",
+        "sha256_hash": "0SmTZgtTRjWa5lXR+XFCqrZcT495jL5XUcJlpMj0uEA=",
+        "updated_at_in_epoch_millis": 1684429230000,
+        "valid_for_in_days": 30,
+        "fields": [
+          "country_iso_code",
+          "country_name",
+          "continent_name",
+          "region_iso_code",
+          "region_name",
+          "city_name",
+          "time_zone",
+          "location"
+        ]
+      },
+      "update_stats": {
+        "last_succeeded_at_in_epoch_millis": 1684866730192,
+        "last_processing_time_in_millis": 317640,
+        "last_failed_at_in_epoch_millis": 1684866730492,
+        "last_skipped_at_in_epoch_millis": 1684866730292
+      }
+    }
+  ]
+}
+```
+
+### Updating an IP2Geo data source
+
+See [Creating the IP2Geo data source]({{site.url}}{{site.baseurl}}/api-reference/ingest-apis/ingest-pipelines/ip2geo/#creating-the-ip2geo-data-source) for endpoints and request field descriptions. 
+
+The following example updates the data source:
+
+```json
+PUT /_plugins/geospatial/ip2geo/datasource/my-datasource/_settings
+{
+    "endpoint": https://geoip.maps.opensearch.org/v1/geolite2-city/manifest.json,
+    "update_interval_in_days": 10
+}
+```
+{% include copy-curl.html %}
+
+### Deleting the IP2Geo data source
+
+ To delete the IP2Geo data source, you must first delete all processors associated with the data source. Otherwise, the request fails. 
+
+The following example deletes the data source:
+
+```json
+DELETE /_plugins/geospatial/ip2geo/datasource/my-datasource
+```
+{% include copy-curl.html %}
+
+## Creating the pipeline
+
+Once the data source is created, you can create the pipeline. The syntax for the `ip2geo` processor is:
+
+```json 
+{
+  "ip2geo": {
+    "field":"ip",
+    "datasource":"my-datasource"
+  }
+}
+```
+{% include copy-curl.html %}
+
+### Configuration parameters
+
+The following table lists the required and optional parameters for the `ip2geo` processor.
+
+| Name | Required | Default | Description |
+|------|----------|---------|-------------|
+| `field` | Required | - | The field that contains the IP address for geographical lookup. |
+| `datasource` | Required | - | The data source name to use to look up geographical information. |
+| `properties` | Optional |  All fields in `datasource`. | The field that controls what properties are added to `target_field` from `datasource`. |
+| `target_field` | Optional | ip2geo | The field that holds the geographical information looked up from the data source. |
+| `ignore_missing` | Optional | false | If `true` and `field` does not exist, the processor quietly exits without modifying the document. |
```

Comment:
The end of the description is a bit oddly worded. Do we just mean "the processor does not modify the document"?

Reply:
Revised to read: If set to `true`, the processor does not modify the document if the field does not exist or is `null`. Default is `false`.